webapp/sources/rudder/rudder-web/src/test/scala/bootstrap/liftweb/checks/migration/TestMigrateCustomProperties.scala

```diff
@@ -0,0 +1,114 @@
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package bootstrap.liftweb.checks.migration
+
+import com.normation.errors
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.facts.nodes.CoreNodeFactRepository
+import com.normation.rudder.facts.nodes.MockLdapFactStorage
+import com.normation.rudder.facts.nodes.NoopGetNodesBySoftwareName
+import com.normation.rudder.facts.nodes.QueryContext
+import com.normation.rudder.facts.nodes.SelectFacts
+import com.normation.rudder.tenants.DefaultTenantService
+
+import com.normation.zio.UnsafeRun
+import org.junit.runner.RunWith
+import org.specs2.mutable.*
+import org.specs2.runner.JUnitRunner
+
+import zio.*
+
+@RunWith(classOf[JUnitRunner])
+class TestMigrateCustomProperties extends Specification {
+
+  val mockLdapFactStorage = new MockLdapFactStorage()
+
+  val factRepo: CoreNodeFactRepository = {
+    (for {
+      tenantService <- DefaultTenantService.make(Nil)
+      repo          <- CoreNodeFactRepository.make(mockLdapFactStorage.nodeFactStorage, NoopGetNodesBySoftwareName, tenantService, Chunk())
+    } yield repo).runNow
+  }
+
+  val migration = new CheckMigrateCustomProperties(mockLdapFactStorage.ldap, factRepo, mockLdapFactStorage.acceptedDIT)
+
+  org.slf4j.LoggerFactory
+    .getLogger("nodes")
+    .asInstanceOf[ch.qos.logback.classic.Logger]
+    .setLevel(ch.qos.logback.classic.Level.TRACE)
+
+  sequential
+
+  "Migrating custom" should {
+    /*
+     * Node1 contains:
+     * - custom properties 'datacenter' and 'from_inv'
+     * - node property 'foo'
+     */
+
+    "find some properties to migrate" in {
+      (migration.findNodeNeedingMigration().runNow must containTheSameElementsAs(List(NodeId("node1")))) and
+      (getNode1PropNamesAsSeenByLdap() must containTheSameElementsAs(List("foo", "datacenter", "from_inv"))) and
+      (getNode1PropNamesAsSeenByRepo() must containTheSameElementsAs(List("foo", "datacenter", "from_inv")))
```

Comment:
I forgot but is this really guaranteed to be true i.e. does the other test that adds `datacenter`, `from_inv` always execute before this one ?

Reply:
it's in the source data